enhancedautoclicker.user.js

```diff
@@ -373,6 +422,121 @@ function autoDungeon() {
     }
 }
 
+function autoAchievement()
+{
+
+    let currentRoute = player.route();
+    let currentRegion = player.region;
+    let currentDungeon = player.town().dungeon?.name;
+    let currentGym = player.town().name;
+
+    //Route
+    if(achievementSelect == 0)
+    {
+        let newRoute = getNextRoute();
+        currentRoute = player.route();
+        if(newRoute && newRoute.number != currentRoute || newRoute.region != currentRegion)
+        {
+            currentRoute = newRoute.number;
+            currentRegion = newRoute.region;
+
+            if(newRoute.region != player.subregion)
+            {
+                player.subregion = newRoute.subRegion;
+            }
+            MapHelper.moveToRoute(newRoute.number, newRoute.region);
+        }
+    }
+    //Gym
+
+    else if(achievementSelect == 1)
+    {
+        //Toggle Auto Gym ON
+        if (gymState === false) {
+            document.getElementById("auto-gym-start").click();
+        }
+
+        let newGym = getNextGym();
+        if(newGym && newGym != currentGym)
+        {
+            currentGym = newGym;
+            player.subregion = TownList[newGym].subRegion;
+            MapHelper.moveToTown(newGym);
+        }
+
+    }
+
+    //Dungeon
+    else if(achievementSelect == 2)
+    {
+        //Toggle Auto Dungeon ON
+        if (dungeonState === false) {
+            document.getElementById("auto-dungeon-start").click();
+        }
+
+        //Move
+        let newDungeon = getNextDungeon();
+        if(newDungeon && newDungeon != currentDungeon)
+        {
+            currentDungeon = newDungeon;
+            player.subregion = TownList[newDungeon].subRegion;
+            MapHelper.moveToTown(newDungeon);
```

Comment:
Does this really work?
When you go from a dungeon to another, does the script keep working?
When playing around with it, I needed to leave the dungeon before going to another one...
So I added this:
```js
                // DungeonRunner.dungeonLeave()
                DungeonRunner.dungeonFinished(true);
                DungeonRunner.fighting(false);
                DungeonRunner.fightingBoss(false);
                MapHelper.moveToTown(DungeonRunner.dungeon.name);
```

If you want to take a look: [AreaDestroyer](<https://github.com/kevingrillet/Userscripts/blob/main/lab/pokeclicker/src/AreaDestroyer.ts>)

Reply:
Thanks for flagging it, I will have a look during the day !

Reply:
I just did a test. 
Are you talking about the next dungeon not starting or the fact the character is not physically moved to the right subregion ?

In Galar, I went to Rose Tower to Dusty Bowl to Courageous Cavern without trouble.
Only issue I had is that my character was not on the Courageous Cavern but on Hammerlocke city when I quit.

![image](https://user-images.githubusercontent.com/15048957/198540641-c1d94341-9b15-4629-a182-23e5b29d6793.png)

Reply:
No, sometimes when you go from a dungeon to another one the script just freezes and autodungeon stop to work.

But If you don't have the problem, it's all good :p

Reply:
Could it be that you beated all Dungeons to more than 500 (assuming you didn't change the value in the code) ?

Reply:
Enter a dungeon, then try to enter it again ;)
`MapHelper.moveToTown(player.town().dungeon.name);`

It end up breaking something

Reply:
Will try ! Thanks for raising it